pandas/core/frame.py

```diff
@@ -1401,11 +1401,45 @@ def to_dict(self, orient="dict", into=dict):
             )
         # GH16122
         into_c = com.standardize_mapping(into)
-        if orient.lower().startswith("d"):
+
+        orient = orient.lower()
```

Comment:
Line 1328 above might also need a change?

Reply:
I would think so - are you interested in submitting a PR to update the documentation?

Reply:
this needs to be removed in a future PR. Check below, there is a  `FutureWarning` that says this will not be accepted anymore, then we can remove this line.

Reply:
Cool, so to confirm, in the future any code like 
`df.to_dict(orient='s')` will stop working?

Reply:
yep, it should